### DISCUSSION TITLE
"App not installed as package appears to be invalid."

### DISCUSSION THREAD

**Discussion by dmunsie** [airsdk/Adobe-Runtime-Support#3002] [Q&A] [answered]
Older SDK installs fine, the latest SDK (overlaid onto Flex) produces the APK without errors, but when trying to install the APK on my phone I get the above message. I'm not using any extensions for this test. Anything I can try? Thank you.

**Comment by FliplineStudios** [comment]
Which was the older AIR SDK that installed fine?  And are you using an IDE for packaging or from the command line?  If it's been a while since you switched SDKs, there are a bunch of requirements for packaging Android builds that you may have to work through with things like Java version and Android SDK and AAB packaging if you haven't already.

**Comment by dmunsie** [comment]
Thx, I was using flexairsdk\4.6.0+33.1.0. It's been a few years since I updated my apps, I've installed the required files (Java jdk-11-0.12), lastest Air SDK, etc. I'm using FlashDevelop/command line for package building. No errors at all, just won't install on my phone. I just did a build using the older tools and it works fine. Strange.... for now I'm just trying to get APKs to build correctly before focusing on bundles.

I do wonder if I should re-install Android Studio. If you were doing a fresh install, should I install a specific version of Android Studio & SDK or just install the latest versions? Thank you.

**Comment by FliplineStudios** [comment, marked as answer]
I'm still using an older Android Studio from 2021 (I only use it for creating adaptive icons, and for downloading the Android SDK for AIR to use), but I'm guessing any version that lets you download the build tools would be fine...  I'm using SDK Build-Tools 33.0.2 for our projects right now, I haven't tried 34 yet myself.

I also just remembered there were some patches for the latest AIR SDK in #2978 for issues with signing Android packages, I'm not sure if they were ever copied over to the main release download since it was right before the holidays -- here's the comment with the latest patch we're using:
https://github.com/airsdk/Adobe-Runtime-Support/issues/2978#issuecomment-1866848279

Might be worth a try with that adt patch first to see if it solves it!

**Reply by dmunsie** [reply]
Thanks! The patch did the trick! :) And I'm using 34 btw, I've created an apk and a bundle, no errors.  Haven't installed the bundle ofcourse, but it was generated with no problem. Thanks for the help, much appreciated.

**Reply by ajwfrost** [reply]
Thanks @FliplineStudios  - we definitely need to get that fix, and a few others, out into a release! Just trying to see if there are any other small + important updates that are needed in it...
